parser/lexer.go

```diff
@@ -926,39 +958,25 @@ func (s *Scanner) lastErrorAsWarn() {
 type reader struct {
 	s string
 	p Pos
-	w int
+	l int
```

Comment:
Why do we need this instead of using `len(s)` directly?

Reply:
`len(s)` is much slower than a const